app/HutchManager/Controllers/AgentsController.cs

```diff
@@ -79,4 +79,9 @@ public async Task<IActionResult> Delete(int id)
     }
     return NoContent();
   }
+  
+  [HttpGet("generate")] //api/generate?isNew=true or false
+  public async Task<ManageAgent> Generate (bool isNew)
```

Comment:
`generate` isn't a great path (or method name) as it sounds like it generates an agent, but it doesn't.

method should probably `GenerateCredentials` or maybe just `GenerateNewSecret` if we move to it only doing replacement secrets.

route will probably just be `/api/agents/{id}/secret` and it should be a POST for an existing agent to get their new secret (and the hash is written to the db at the same time.